tests/models/test_image.py

```diff
@@ -97,7 +100,11 @@ def test_image_archive_singlefile(demo_image, provenance_base, archive):
 
     try:
         with SmartSession() as session:
-            demo_image.provenance = provenance_base
+            # SQLAlchemy can be a real PITA.  Have to merge things into
+            # it (plus see below before session.add(demo_image)) to
+            # avoid having it refuse to do things.
+            demo_image.provenance = session.merge( provenance_base )
+            exposure.provenance = session.merge( exposure.provenance )
```

Comment:
Try using `recursive_merge` from `base.py`. I think it may solve these problems (after eating sh*t from this same issue I added something that hopefully solves it).